### DISCUSSION TITLE
Gradle and self-signed certificate

### DISCUSSION THREAD

**Discussion by MageFroh** [renovatebot/renovate#17384] [Request Help]
Hi there,

I'm using the renovate Docker container in a self-hosted environment.
Some of our projects are using Gradle, and their dependencies are hosted in an internal Maven registry.
That maven registry uses self-signed certificates.

Taking the end of https://docs.renovatebot.com/examples/self-hosting/#self-signed-tlsssl-certificates as an example, we have written a `Dockerfile` that extends `renovate/renovate`.
It is in charge of adding the local CA certificate in the OS trust store and in the JRE trust store.

It used to look like this:
```
FROM renovate/renovate

USER root

COPY local-ca.crt /usr/local/share/ca-certificates/
RUN update-ca-certificates

ENV NODE_EXTRA_CA_CERTS=/usr/local/share/ca-certificates/local-ca.crt

#### This is the important bit that used to update the JRE trust store
RUN source /usr/local/env.d/java.sh ; \
    $JAVA_HOME/bin/keytool -import -trustcacerts -alias local-ca -file /usr/local/share/ca-certificates/local-ca.crt -keystore $JAVA_HOME/lib/security/cacerts -storepass changeit

USER 1000
```

However, it does not work anymore: the `java.sh` script seems to have been removed from recent images (as in less than 7 days ago).

Do you have a recommendation about how to update the java trust store with a local CA certificate?

I could change the line to the following, but this feels brittle:
```
RUN /opt/buildpack/tools/java/*/bin/keytool -import -trustcacerts -alias local-ca -file /usr/local/share/ca-certificates/local-ca.crt -keystore /opt/buildpack/tools/java/*/lib/security/cacerts -storepass changeit
```

Thanks,
Benoit

**Comment by rarkins** [comment]
Can you try the instructions at https://github.com/containerbase/buildpack/blob/main/docs/custom-root-ca.md#java-1 ?

**Reply by MageFroh** [reply]
Thanks @rarkins - That seems to work :)

**Reply by rarkins** [reply]
Great! Would you mind adding what you learned to https://github.com/renovatebot/renovate/blob/main/docs/usage/examples/self-hosting.md via a PR?

**Reply by MageFroh** [reply]
Sure, I'll open one soon
